src/Contracts/ConfigInterface.php

```diff
@@ -5,10 +5,16 @@
 /**
  * parameters in constructor are not require
  */
-interface ConfigContract
+interface ConfigInterface
 {
     public function __construct();
 
+    /**
+     * @param array<string, mixed> $an_array
+     * @return static
+     */
+    public static function __set_state(array $an_array): object;
+
     /**
      * @param static $object
      */
```

Comment:
říkám si jestli metoda `public function initFromDefaultConfig(ConfigInterface $object): void` bude fungovat, pokud se nakešuje do konfigurace pomocí __set_state, to by ty předávané objekty taky museli implementovat __set_state. Zatím to není nijak ošetřeno a pravděpodobně to zdechne.